basculehttp/log.go

```diff
@@ -69,11 +73,48 @@ func sanitizeHeaders(headers http.Header) (filtered http.Header) {
 func SetLogger(logger *zap.Logger) alice.Constructor {
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			r = r.WithContext(sallust.With(r.Context(),
-				logger.With(
-					zap.Reflect("requestHeaders", sanitizeHeaders(r.Header)), //lgtm [go/clear-text-logging]
-					zap.String("requestURL", r.URL.EscapedPath()),
-					zap.String("method", r.Method))))
+			var tid string
+
+			if tid = r.Header.Get(transaction.HeaderWPATID); tid == "" {
+				tid = transaction.GenTID()
+			}
+
+			var satClientID = "N/A"
+
+			// retrieve satClientID from request context
+			if auth, ok := bascule.FromContext(r.Context()); ok {
+				satClientID = auth.Token.Principal()
+			}
+
+			var source string
+			host, _, err := net.SplitHostPort(r.RemoteAddr)
+			if err != nil {
+				source = r.RemoteAddr
+			} else {
+				source = host
+			}
+
+			logger = logger.With(
+				zap.Reflect("requestHeaders", sanitizeHeaders(r.Header)), //lgtm [go/clear-text-logging]
+				zap.String("requestURL", r.URL.EscapedPath()),
+				zap.String("method", r.Method),
+				zap.Reflect("request", transaction.Request{
+					Address: source,
+					Path:    r.URL.Path,
+					Query:   r.URL.RawQuery,
+					Method:  r.Method},
+				),
```

Comment:
We don't want to import the tr1d1um `transaction` package.  Also, I don't think this is a great way to store fields - I would rather store each one in its own key-value pair (I think this is more performant, since we know all the types).  Method and URL are kind of here twice.  I would suggestion something like below:
```suggestion
				zap.Reflect("request.headers", sanitizeHeaders(r.Header)), //lgtm [go/clear-text-logging]
				zap.String("request.url", r.URL.EscapedPath()),
				zap.String("request.method", r.Method),
				zap.String("request.address", source),
				zap.String("request.query", r.URL.RawQuery),
```